src/wrappers/themis/Obj-C/objcthemis/smessage.m

```diff
@@ -136,36 +140,73 @@ - (nullable NSData *)wrapData:(nullable NSData *)message error:(NSError * __auto
 }
 
 
-- (nullable NSData *)unwrapData:(nullable NSData *)message error:(NSError * __autoreleasing *)error {
+- (nullable NSData *)unwrapData:(nullable NSData *)message error:(NSError *__autoreleasing *)error {
     size_t unwrappedMessageLength = 0;
+    TSErrorType result = TSErrorTypeFail;
+
+    switch (self.mode) {
```

Comment:
a bit of refactoring here – previously this function didn't check the mode, now we do (and check keys)